src/testdrive/src/action/kafka/verify.rs

```diff
@@ -109,10 +109,23 @@ async fn get_topic(
     topic_field: &str,
     state: &mut State,
 ) -> Result<String, anyhow::Error> {
-    let query = format!("SELECT {} FROM mz_catalog_names JOIN mz_kafka_sinks ON global_id = sink_id WHERE name = $1", topic_field);
+    let query = format!(
+        "SELECT {} FROM mz_sinks JOIN mz_kafka_sinks \
+        ON mz_sinks.id = mz_kafka_sinks.sink_id \
+        JOIN mz_schemas s ON s.id = mz_sinks.schema_id \
+        LEFT JOIN mz_databases d ON d.id = s.database_id \
+        WHERE d.name = $1 \
+        AND s.name = $2 \
+        AND mz_sinks.name = $3",
+        topic_field
+    );
+    let sink_fields: Vec<&str> = sink.split('.').collect();
     let result = state
         .pgclient
-        .query_one(query.as_str(), &[&sink])
+        .query_one(
+            query.as_str(),
+            &[&sink_fields[0], &sink_fields[1], &sink_fields[2]],
```

Comment:
@mjibson I couldn't figure out a cleaner way to get this working in Rust - the parameter needs to be a slice with elements of type `tokio_postgres::types::ToSql`. If you construct the slice within the function call it auto-converts the strs into this type, but if you pre-create the slice and pass it in, I get an error.

Reply:
I think you might be able to use `Vec<&dyn ToSql>` as the type annotation on `sink_fields`, and then you’d be able to pass in `sink_fields` here directly. If not… well that’s just how it goes in Rust, and this seems fine!

Reply:
Sadly this was not the case :(
On using that type I get `value of type Vec<&dyn tokio_postgres::types::ToSql> cannot be built from std::iter::Iterator<Item=&str>`.
And the `query_one` function doesn't accept it anyways, with error `expected slice, found struct Vec`.
I'm fine to leave it as-is unless anyone has better suggestions.

Reply:
No better suggestions!